script.js

```diff
@@ -1,4 +1,251 @@
+        cardName = "king";
+      }
+      var card = {
+        name: cardName,
+        suit: currentSuit,
+        rank: rankCounter,
+      };
+      cardDeck.push(card);
+      rankCounter += 1;
+    }
+    suitIndex += 1;
+  }
+  return cardDeck;
+}
+
+function getRandomIndex(max) {
+  return Math.floor(Math.random() * max);
+}
+
+function shuffleCards(cardDeck) {
+  var currentIndex = 0;
+  while (currentIndex < cardDeck.length) {
+    var randomIndex = getRandomIndex(cardDeck.length);
+    var randomCard = cardDeck[randomIndex];
+    var currentCard = cardDeck[currentIndex];
+    cardDeck[currentIndex] = randomCard;
+    cardDeck[randomIndex] = currentCard;
+    currentIndex = currentIndex + 1;
+  }
+  return cardDeck;
+}
+
+function dealCards(handArray, cardDeck, noOfCards) {
+  for (var counter = 0; counter < noOfCards; counter += 1) {
+    handArray.push(cardDeck.pop());
+  }
+  return handArray;
+}
+
+function isBlackjack(handArray) {
+  return (
+    (handArray[0].name === "ace" && handArray[1].rank >= 10) ||
+    (handArray[1].name === "ace" && handArray[0].rank >= 10)
+  );
+}
+
+function calculateScore(handArray) {
+  var score = 0;
+  for (var counter = 0; counter < handArray.length; counter += 1) {
+    score = score + getCardScore(handArray[counter]);
+  }
+  //if there are aces, update ace value in score. if total hand value is >21, update ace value from 11 to 1
+  for (var counter = 0; counter < handArray.length; counter += 1) {
+    if (handArray[counter].name === "ace") {
+      score = updateAceValueinScore(score);
+    }
+  }
+  return score;
+}
+
+function getCardScore(card) {
+  score = 0;
+  if (card.name === "king" || card.name === "queen" || card.name === "jack") {
+    score = 10;
+  } else if (card.name === "ace") {
+    score = 11;
+  } else {
+    score = card.rank;
+  }
+  return score;
+}
+
+function updateAceValueinScore(score) {
+  //if total value of hand is >21, update ace value from 11 to 1
+  if (score > 21) {
+    score = score - 10;
+  }
+  return score;
+}
+
+function displayPlayerHandAndScore(player, handArray, score, gameMode) {
+  message = "";
+
+  if (gameMode === "startGame") {
+    message = `${player}'s Hand: ${listCards(player, handArray, gameMode)}`;
+  } else {
+    message = `${player}'s Hand: ${listCards(
+      player,
+      handArray,
+      gameMode
+    )} Score: ${score}`;
+  }
+
+  if (isBust(handArray)) {
+    message = message + " (BUST!)";
+  }
+  message = message + "<br><br>";
+  return message;
+}
+
+function listCards(player, handArray, gameMode) {
+  var cardList = "";
+  if (player === "Dealer" && gameMode === "startGame") {
+    cardList =
+      cardList +
+      `- ${handArray[0].name} of ${handArray[0].suit}<br> - (Hidden Card)<br>`;
+  } else {
+    for (i = 0; i < handArray.length; i += 1) {
+      cardList =
+        cardList + `- ${handArray[i].name} of ${handArray[i].suit}<br>`;
+    }
+  }
+  return `<br> ${cardList}`;
+}
+
+function isBust(handArray) {
```

Comment:
Nice function! Could probably combine it into one line
`return calculateScore(handArray) > 21`